buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy

```diff
@@ -665,6 +665,12 @@ class ClusterFormationTasks {
     static Task configureExecTask(String name, Project project, Task setup, NodeInfo node, Object[] execArgs) {
         return project.tasks.create(name: name, type: LoggedExec, dependsOn: setup) { Exec exec ->
             exec.workingDir node.cwd
+            if (project.runtimeJavaHome.equals(project.compilerJavaHome)) {
```

Comment:
The purpose of this conditional is to allow testing with Java home set. This is just checking whether runtime Java home was set, since if it wasn't it would be the same as the compiler Java. If we compared to the bundled version, we would likely never test the bundled jdk as devs aren't always on the same version on their machines as we are bundling.

Reply:
I wasn't thinking about matching the version exactly, just the major, since that's what our matrix CI jobs also care about and what's included in the reproduction lines. 
I don't think the conditional would behave differently, but it would convey intent better without relying on an implementation detail in the build plugin.

Reply:
The intent is to test the default/common behavior by default. That is, to test the bundled jdk. If runtime java home is set, this tests whatever is there, whether is is an external 11 or 8. That seems straightforward to me.

Reply:
I wouldn't have a problem with that, but we are not testing directly if `ES_RUNTIME_JAVA` is passed in, but rely on knowledge about how this is passed along instead. I think that these are worth avoiding because it makes the build harder to maintain. In other words looking at just this line and ignoring everything else you know about the build would make you wonder why the java used for compilation would matter in weather we do  or do not use the bounded JDK ?

Reply:
@atorok Tests are passing now. Do you have a concrete suggestion here? I do not want to compare anything about runtimeJavaHome to the version being bundled. Just because they are the same major version of java does not mean they are the same.

Reply:
I added an extension property to indicate whether runtime java home is set in 
29ec298